mlrun/runtimes/function.py

```diff
@@ -527,3 +519,106 @@ def get_auth_filled_platform_dashboard_url(dashboard: str) -> str:
         mlconf.nuclio_dashboard_url or "nuclio-api",
         parsed_dbpath.hostname[parsed_dbpath.hostname.find(".") :],
     )
+
+
+def get_fullname(name, project, tag):
+    if project:
+        name = "{}-{}".format(project, name)
+    if tag:
+        name = "{}-{}".format(name, tag)
+    return name
+
+
+def deploy_nuclio_function(function: RemoteRuntime, dashboard="", watch=False):
+    function.set_config("metadata.labels.mlrun/class", function.kind)
+    env_dict = {get_item_name(v): get_item_name(v, "value") for v in function.spec.env}
+    for key, value in function._get_runtime_env().items():
+        env_dict[key] = value
+    spec = nuclio.ConfigSpec(env=env_dict, config=function.spec.config)
+    spec.cmd = function.spec.build.commands or []
+    project = function.metadata.project or "default"
+    tag = function.metadata.tag
+    handler = function.spec.function_handler
+
+    # In Nuclio 1.6.0 default serviceType changed to "ClusterIP", make sure we're using NodePort
+    spec.set_config("spec.serviceType", "NodePort")
+    if function.spec.readiness_timeout:
+        spec.set_config("spec.readinessTimeoutSeconds", function.spec.readiness_timeout)
+    if function.spec.resources:
+        spec.set_config("spec.resources", function.spec.resources)
+    if function.spec.no_cache:
+        spec.set_config("spec.build.noCache", True)
+    if function.spec.replicas:
+        spec.set_config("spec.minReplicas", function.spec.replicas)
+        spec.set_config("spec.maxReplicas", function.spec.replicas)
+    else:
+        spec.set_config("spec.minReplicas", function.spec.min_replicas)
+        spec.set_config("spec.maxReplicas", function.spec.max_replicas)
+
+    dashboard = get_auth_filled_platform_dashboard_url(dashboard)
```

Comment:
No need for this anymore (since this is running in the API context) the helm chart already configures the `config.nuclio_dashboard_url` to the correct address, simply read from config

Reply:
done